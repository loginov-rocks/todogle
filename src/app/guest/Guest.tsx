import { Box, Button } from '@material-ui/core';
import * as React from 'react';

import Logo from '../../components/logo/Logo';
import gapi from '../../services/gapi';

import googleIcons from './google.svg';
import styles from './Guest.module.css';

const Guest = () => {
  const handleClick = () => {
    gapi.signIn();
  };

  return (
    <div className={styles.root}>
      <Box mb={3}>
        <Logo />
      </Box>
      <Button
        onClick={handleClick}
        size="large"
        startIcon={<img src={googleIcons} alt="Google" />}
        variant="outlined"
      >
        Sign In
      </Button>
    </div>
  );
};

export default Guest;
