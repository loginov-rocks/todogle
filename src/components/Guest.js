import React from 'react';

import gapi from '../services/gapi';

export default class Guest extends React.Component {
  handleClick() {
    gapi.signIn();
  }

  render() {
    return <button onClick={this.handleClick}>Sign In</button>;
  }
}
