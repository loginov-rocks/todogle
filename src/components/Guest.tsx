import { Button } from '@material-ui/core';
import * as React from 'react';

import gapi from '../services/gapi';

export default class Guest extends React.Component {
  handleClick() {
    gapi.signIn();
  }

  render() {
    return (
      <Button onClick={this.handleClick} variant="contained">Sign In</Button>
    );
  }
}
