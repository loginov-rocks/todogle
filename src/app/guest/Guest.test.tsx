import { shallow } from 'enzyme';
import * as React from 'react';

import Guest from './Guest';

it('matches snapshot', () => {
  const wrapper = shallow(
    <Guest />,
  );

  expect(wrapper).toMatchSnapshot();
});
