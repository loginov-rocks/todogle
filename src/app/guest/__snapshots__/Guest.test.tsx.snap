// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="root"
>
  <Styled(MuiBox)
    mb={3}
  >
    <Logo />
  </Styled(MuiBox)>
  <WithStyles(ForwardRef(Button))
    onClick={[Function]}
    size="large"
    startIcon={
      <img
        alt="Google"
        src="google.svg"
      />
    }
    variant="outlined"
  >
    Sign In
  </WithStyles(ForwardRef(Button))>
</div>
`;
